clang/lib/Sema/SemaOverload.cpp

```diff
@@ -2088,6 +2091,57 @@ static bool IsVectorElementConversion(Sema &S, QualType FromType,
   return false;
 }
 
+/// Determine whether the conversion from FromType to ToType is a valid
+/// matrix conversion.
+///
+/// \param ICK Will be set to the matrix conversion kind, if this is a matrix
+/// conversion.
+static bool IsMatrixConversion(Sema &S, QualType FromType, QualType ToType,
+                               ImplicitConversionKind &ICK,
+                               ImplicitConversionKind &ElConv, Expr *From,
+                               bool InOverloadResolution, bool CStyle) {
+  // The non HLSL Matrix conversion rules are not clear.
```

Comment:
I'd probably instead write this as:
```suggestion
  // Implicit conversions for matrices are an HLSL feature not present in C/C++.
```

I suspect that HLSL's row and column based truncation would be quite inefficient on most CPUs, so there would probably be a more constrained way to express conversions ideally not implicitly.